src/pages/Calendar/Calendar-Day.vue

```diff
@@ -0,0 +1,393 @@
+        },
+        loadingKey: 'loadingKey'
+      })
+      const ongoingRuns = await this.$apollo.query({
+        query: require('@/graphql/Calendar/calendar-day-ongoing-flow-runs.gql'),
+        variables: {
+          startTime: this.start,
+          endTime: this.date,
+          flowId: this.flowId
+        },
+        loadingKey: 'loadingKey'
+      })
+      const allRuns = [
+        ...(finishedRuns?.data?.flow_run ? finishedRuns.data.flow_run : []),
+        ...(scheduledRuns?.data?.flow_run ? scheduledRuns.data.flow_run : []),
+        ...(runningRuns?.data?.flow_run ? runningRuns.data.flow_run : []),
+        ...(ongoingRuns?.data?.flow_run ? ongoingRuns.data.flow_run : [])
+      ]
+      const uniqueRuns = [...new Set(allRuns)]
+
+      const updatedRuns = uniqueRuns.map(flowRun => {
+        const diff = new Date(flowRun.end_time) - new Date(flowRun.start_time)
+        const addedTime = this.addTime(flowRun.start_time, 2, 'm')
+
+        flowRun.start = !flowRun.start_time
+          ? this.formatCalendarTime(flowRun.scheduled_start_time)
+          : this.formatCalendarTime(flowRun.start_time)
+
+        flowRun.end =
+          flowRun.start_time && !flowRun.end_time
+            ? this.formatCalendarTime(new Date())
+            : diff < 60000
+            ? addedTime
+            : flowRun.start_time < this.date &&
+              flowRun.end_time?.split('T')[0] > this.date
+            ? this.formatCalendarTime(new Date())
+            : this.formatCalendarTime(flowRun.end_time)
+
+        flowRun.timed =
+          flowRun.start_time < this.date
+            ? !flowRun.end_time
+              ? false
+              : flowRun.end_time.split('T')[0] > this.date
+              ? false
+              : true
+            : true
+
+        flowRun.category = flowRun.flow_id
+        return flowRun
+      })
+
+      this.intervalHeight =
+        this.type === 'day'
+          ? uniqueRuns.length < 100
+            ? 100
+            : uniqueRuns.length > 1500
+            ? uniqueRuns.length / 2
+            : uniqueRuns.length
+          : 100
+
+      this.gettingRuns = false
+
+      this.scheduleBanner =
+        this.flow?.is_schedule_active &&
+        this.upcoming.length > 8 &&
+        !this.closeBanner
+          ? true
+          : !this.closeBanner
+          ? new Date(this.date) > new Date()
+          : false
+      return updatedRuns
+    },
+    eventColor(event) {
+      return event.state ? event.state : 'primary'
+    },
+    striped(event) {
+      return event.state === 'Scheduled'
+        ? 'striped'
+        : event.state === 'Submitted'
+        ? 'darker'
+        : ''
+    },
+    handleEventClick({ nativeEvent, event }) {
+      const open = () => {
+        this.selectedEvent = event
+        this.selectedElement = nativeEvent.target
+        this.selectedOpen = true
+      }
+      if (this.selectedOpen) {
+        this.selectedOpen = false
+        setTimeout(open, 10)
```

Comment:
As a matter of best practice, it'd be good to capture this timeout id and clear it at the start of this handler